openpype/hosts/maya/plugins/publish/extract_workfile_xgen.py

```diff
@@ -0,0 +1,250 @@
+import os
+import shutil
+import copy
+
+from maya import cmds
+
+import pyblish.api
+from openpype.hosts.maya.api.lib import extract_alembic
+from openpype.pipeline import publish
+from openpype.lib import StringTemplate
+
+
+class ExtractWorkfileXgen(publish.Extractor):
+    """Extract Workfile Xgen.
+
+    When submitting a render, we need to prep Xgen side car files.
+    """
+
+    # Offset to run before workfile scene save.
+    order = pyblish.api.ExtractorOrder - 0.499
+    label = "Extract Workfile Xgen"
+    families = ["workfile"]
+    hosts = ["maya"]
+
+    def get_render_max_frame_range(self, context):
+        """Return start to end frame range including all renderlayers in
+        context.
+
+         This will return the full frame range which includes all frames of the
+         renderlayer instances to be published/submitted.
+
+         Args:
+            context (pyblish.api.Context): Current publishing context.
+
+         Returns:
+            tuple or None: Start frame, end frame tuple if any renderlayers
+                found. Otherwise None is returned.
+
+         """
+
+        def _is_active_renderlayer(i):
+            """Return whether instance is active renderlayer"""
+            if not i.data.get("publish", True):
+                return False
+
+            is_renderlayer = (
+                "renderlayer" in i.data.get("families", []) or
+                i.data["family"] == "renderlayer"
+            )
+            return is_renderlayer
+
+        start_frame = None
+        end_frame = None
+        for instance in context:
+            if not _is_active_renderlayer(instance):
+                # Only consider renderlyare instances
+                continue
+
+            render_start_frame = instance.data["frameStart"]
+            render_end_frame = instance.data["frameStart"]
+
+            if start_frame is None:
+                start_frame = render_start_frame
+            else:
+                start_frame = min(start_frame, render_start_frame)
+
+            if end_frame is None:
+                end_frame = render_end_frame
+            else:
+                end_frame = max(end_frame, render_end_frame)
+
+        if start_frame is None or end_frame is None:
+            return
+
+        return start_frame, end_frame
+
+    def process(self, instance):
+        transfers = []
+
+        # Validate there is any palettes in the scene.
+        if not cmds.ls(type="xgmPalette"):
+            self.log.debug(
+                "No collections found in the scene. Skipping Xgen extraction."
+            )
+            return
+        else:
+            import xgenm
```

Comment:
I don't feel like this really needs to be in an `else` statement since we return in the if statement anyway. 
We could just do this import below the `get_render_max_frame_range` call below, no?

Reply:
yeah, good point. Going outside the if statement.